app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,24 @@ def index
 
   def show
     @post = Post.find_by(id: params[:id])
+    @user = @post.author_id
+  end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    post = Post.new(author: current_user, text: params[:post][:text], title: params[:post][:title],
```

Comment:
- [ ] [OPTIONAL] For creating any model attributes, always use strong parameters. like you have used comment_params in comments_controller.

Reply:
- [ ] [optional] @biruk-bereka great work implementing the changes requested, like the previous reviewer suggested I would strongly recommend the use of strong parameters, this increases the security of data sent using forms.